### PR TITLE
ensure backend up-to-date and backend core refactorings

### LINKED ISSUE #350: upgrade to php 7.1 and symfony 3.2
### Description

there are various RCs for 7.1 and it won't take too much time until PHP 7.1 is ready.
I'm in favor of upgrading just for the sake of dropping PHP 7.0 as the lack of nullable types is just annoying in certain cases.

### Tasks

- [x] ensure latest packages (PHP 7.1, Symfony 3.2, PHPUnit 5.7)
- [x] keep code-base up-to-date
  - [x] use types everywhere (e.g. void or nullable if not already present)
  - [x] drop `list()` usage for new `[]` syntax
  - [x] make constants private when possible
  - [x] use `Closure::fromCallable` rather than `$that = &$this;`. (In the `OnlineUserIdCluster` a dependency needs to be injected into a callback. This can be simplified to avoid these dirty `$that` hacks)
- [x] keep phpunit tests up-to-date
  - [x] use `createMock` everywhere
- [x] keep symfony-related code up-to-date
  - [x] use `ConstraintValidatorTestCase` rather than legacy `AbstractConstraintValidatorTest`

### PR DESCRIPTION
**Pull request by Ma27**
### Overview

this PR aims to ensure all backend dependencies to be up-to-date and that PHP 7.1 features are actively used (enhanced type system).
The backend needed some refactoring to comply with the new typesystem in a better way.
When merging this, I'll split the commits in a better way to segregate refactorings from type system fixes.

### Related issues

resolves #350

### CODE REVIEW

File: src/AppBundle/Model/Ip/Value/IpLocation.php [hunk 1]
```diff
@@ -130,4 +142,14 @@ public function getLongitude(): float
     {
         return $this->longitude;
     }
+
+    /**
+     * Checks if the VO is actually empty.
+     *
+     * @return bool
+     */
+    public function isEmpty()
```
Comment (Ma27) [inline src/AppBundle/Model/Ip/Value/IpLocation.php]: missing boolean return type

**Review by Ma27 — COMMENTED** [review]
otherwise looks good to me.

**Review by benbieler — COMMENTED** [review]
Seems fine to me too. But we should probably change the commit message guidelines (https://github.com/Sententiaregum/Sententiaregum/blob/master/.github/CONTRIBUTING.md#commit-message)

**Comment by Ma27** [comment]
yep, you're right. Planned to do that before merging :)

**Comment by Ma27** [comment]
ok, build is green.
Will rebuild the commit objects to make the history way better and then this can be merged.

**Pull request merged**